llvm/lib/Transforms/IPO/Attributor.cpp

```diff
@@ -1749,6 +1749,10 @@ bool Attributor::checkForAllCallees(
   return Pred(Callees.getArrayRef());
 }
 
+bool noRevisitUser(const User *user) {
```

Comment:
Is there a better name for this function as this is not really checking the revisiting of a node as the Visited map does. But rather checking for uses that are not instructions.

Reply:
Maybe `markAsVisited` would work better?

Regarding your initial point, I do plan on squashing and merging this commit once it is approved.

Reply:
`canMarkAsVisited` seems fine 
I would even recommend, `checkIfPhiOrNotInst`